## Method 4: reference distribution based on random sampling, internal estimate of  𝜎 
Let's recall the case study. We have 10 data points for method A and 10 data points for method B.

The means are different: 84.24 for method A vs. 85.54 for method B.

The reference distribution has 210 data points with a mean of 84.12

Question: are 84.24 and 85.54 statistically different based on the reference distribution?

To answer with method 4, we continue using the assumption that each individual observation is as if obtained by random sampling from a normal population with common standard deviation.

But this time, instead of deriving the standard deviation from the 210 reference data points, we derive it from the sample itself! In real life, we often do not have access to reference set and only have the samples so very important to be able to extract the maximum of information from the samples themselves.

We compute the same statistic as with method 3 but the difference is that instead of $\sigma$, we will now use s which is the pooled estimate of the sample variance based on sample A and sample B. Therefore, the z statistic becomes a t statistic following t distribution.

$$t = \frac{(\bar{y_B}-\bar{y_A}) - (\eta_B-\eta_A)}{s\sqrt{1/n_A+1/n_B}}$$

Because the pooled variance has $n_A + n_B -2$ (=18) degrees of freedom, so has the t statistic as well.

In [4]:
import pandas as pd
import numpy as np
import math
import scipy.stats as st
import matplotlib.pyplot as plt
%config Completer.use_jedi = False
pd.set_option('display.max_rows', 500)
y_210 = pd.read_excel('yield 210.xlsx')
y_AB = pd.read_excel('yield 20.xlsx')

In [20]:
yBbar_m_yAbar = 1.30
etaB_m_etaA = 0
nA, nB = 10, 10
sqa = (1/(nA-1))*((y_AB[y_AB['method'] == 'A']['yield']-y_AB[y_AB['method'] == 'A']['yield'].mean(axis=0))**2).sum()
sqb = (1/(nB-1))*((y_AB[y_AB['method'] == 'B']['yield']-y_AB[y_AB['method'] == 'B']['yield'].mean(axis=0))**2).sum()
sq = (1/(nA+nB-2))*(((y_AB[y_AB['method'] == 'A']['yield']-y_AB[y_AB['method'] == 'A']['yield'].mean(axis=0))**2).sum()+
                 ((y_AB[y_AB['method'] == 'B']['yield']-y_AB[y_AB['method'] == 'B']['yield'].mean(axis=0))**2).sum())
t = (yBbar_m_yAbar - etaB_m_etaA)/math.sqrt(sq*(1/nA+1/nB))
p_value = 1-st.t.cdf(t,df=nA+nB-2,loc=0,scale=1)
print('sqa: {:.02f}, sqb: {:.02f}, sq: {:.02f}'.format(sqa,sqb,sq))
print('t: {:.02f}, p value: {:.03f}'.format(t,p_value))

sqa: 8.42, sqb: 13.32, sq: 10.87
t: 0.88, p value: 0.195


Once again, the p value is high and we cannot reject the null hypothesis, confirming the weakness of random sampling assumption coupled with variance estimated from the sample itself.

Notice how the significance of 19.5% differs significantly from the much lower numbers obtained with method 1 (4.7%) and method 2 (2.8%) and even method 3 (15.7%). The more we add assumptions to compensate for the lack of external reference data, the weaker our statistic becomes to detect changes.